src/ESLintLinter.php

```diff
@@ -163,7 +163,11 @@ protected function parseLinterOutput($path, $err, $stdout, $stderr) {
         $message = new ArcanistLintMessage();
         $message->setPath($file['filePath']);
         $message->setSeverity($this->mapSeverity($offense['severity']));
-        $message->setName($offense['ruleId'] || 'unknown');
+        if (isset($offense['ruleId'])) {
```

Comment:
libphutil has a helper function for this pattern:

https://github.com/phacility/libphutil/blob/39ed96cd818aae761ec92613a9ba0800824d0ab0/src/utils/utils.php#L36

So maybe:

```php
$message->setName(idx($offense, ‘ruleId’, $this->getLinterName());
```

Reply:
`idx($offense, ‘ruleId’, $this->getLinterName()` will return `null` if `$offense['ruleId']` is `null`. I can't think of a case in which the `ruleId` key exists and its value is set to `null`, but for all I know there might be one. As such, I slightly prefer the `isset` check that I have used in this PR.

Reply:
(Because, to clarify, `$message->setName(null)` will result in I think a runtime exception. It's been a couple of days since I repro'ed this case, but I believe that was the problem. The bug with `setName(null)` is the exact same bug that led to #22.)

Reply:
Won't the `isset()` check you have here also succeed when `$offense['ruleId']` is `null`? I think you'd need `!empty($offense['ruleId'])` to guard against that case.

The equivalent phutil construct would probably use [`nonempty()`](https://github.com/phacility/libphutil/blob/39ed96cd818aae761ec92613a9ba0800824d0ab0/src/utils/utils.php#L745):

```php
$message->setName(nonempty($offense['ruleId'], $this->getLinterName()));
```